mito-ai/src/utils/websocket/websocketClient.ts

```diff
@@ -268,6 +289,66 @@ export class CompletionWebsocketClient implements IDisposable {
     await this._ready.promise;
   }
 
+  /**
+   * Maximum number of reconnection attempts
+   */
+  private _maxReconnectAttempts = 5;
+
+  /**
+   * Current reconnection attempt
+   */
+  private _reconnectAttempt = 0;
+
+  /**
+   * Attempt to reconnect the websocket with exponential backoff
+   */
+  async reconnect(forceReset: boolean = true): Promise<void> {
+    if (this._isDisposed) {
+      throw new Error('Client is disposed');
+    }
+    
+    // Reset _reconnectAttempt if this is a manual reconnect
+    if (forceReset) {
+      this._reconnectAttempt = 0;
+    }
+    
+    if (this._reconnectAttempt >= this._maxReconnectAttempts) {
+      throw new Error(`Failed to reconnect after ${this._maxReconnectAttempts} attempts`);
+    }
+    
+    // Clean up any existing socket
+    if (this._socket) {
+      this._socket.close();
+      this._socket = null;
+    }
+    
+    // Calculate delay using exponential backoff
+    // First attempt is immediate, then 1s, 2s, 4s, 8s
+    const delay = this._reconnectAttempt === 0 ? 0 : Math.pow(2, this._reconnectAttempt - 1) * 1000;
+    
+    if (delay > 0) {
+      console.log(`Attempting to reconnect in ${delay / 1000}s (attempt ${this._reconnectAttempt + 1}/${this._maxReconnectAttempts})...`);
+      await new Promise(resolve => setTimeout(resolve, delay));
+    }
+    
+    this._reconnectAttempt++;
+    
+    try {
+      // Reinitialize connection
+      await this._initialize();
+      // Reset reconnect attempt counter on success
+      this._reconnectAttempt = 0;
+    } catch (error) {
+      console.error(`Reconnection attempt ${this._reconnectAttempt} failed:`, error);
+      // Try again recursively with exponential backoff if we haven't exceeded the limit
+      if (this._reconnectAttempt < this._maxReconnectAttempts) {
```

Comment:
We already have this check above on line 315, so I think we can just call the this.reconnect(false) again and let the function handle it